packages/contracts/test/exchange/internal.ts

```diff
@@ -0,0 +1,329 @@
+            denominator: BigNumber,
+            target: BigNumber,
+        ): Promise<boolean> {
+            if (numerator.greaterThan(MAX_UINT256)) {
+                throw invalidOpcodeErrorForCall;
+            } else if (denominator.greaterThan(MAX_UINT256)) {
+                throw invalidOpcodeErrorForCall;
+            } else if (denominator.eq(new BigNumber(0))) {
+                throw invalidOpcodeErrorForCall;
+            } else if (target.greaterThan(MAX_UINT256)) {
+                throw invalidOpcodeErrorForCall;
+            }
+            const product = numerator.mul(target);
+            const remainder = product.mod(denominator);
+            if (remainder.eq(new BigNumber(0))) {
+                return false;
+            }
+            if (product.greaterThan(MAX_UINT256)) {
+                throw overflowErrorForCall;
+            }
+            if (product.eq(new BigNumber(0))) {
+                throw invalidOpcodeErrorForCall;
+            }
+            const remainderTimes1000000 = remainder.mul(new BigNumber('1000000'));
+            if (remainderTimes1000000.greaterThan(MAX_UINT256)) {
+                throw overflowErrorForCall;
+            }
+            const errPercentageTimes1000000 = remainderTimes1000000.dividedToIntegerBy(product);
+            return errPercentageTimes1000000.greaterThan(new BigNumber('1000'));
+        }
+        async function testIsRoundingErrorAsync(
+            numerator: BigNumber,
+            denominator: BigNumber,
+            target: BigNumber,
+        ): Promise<boolean> {
+            return testExchange.publicIsRoundingError.callAsync(numerator, denominator, target);
+        }
+        await testCombinatoriallyWithReferenceFuncAsync(
+            'isRoundingError',
+            referenceIsRoundingErrorAsync,
+            testIsRoundingErrorAsync,
+            [uint256Values, uint256Values, uint256Values],
+        );
+    });
+
+    describe('updateFilledState', async () => {
+        // Note(albrow): Since updateFilledState modifies the state by calling
+        // sendTransaction, we must reset the state after each test.
+        beforeEach(async () => {
+            await blockchainLifecycle.startAsync();
+        });
+        afterEach(async () => {
+            await blockchainLifecycle.revertAsync();
+        });
+        async function referenceUpdateFilledStateAsync(
+            takerAssetFilledAmount: BigNumber,
+            orderTakerAssetFilledAmount: BigNumber,
+            // tslint:disable-next-line:no-unused-variable
+            orderHash: string,
+        ): Promise<BigNumber> {
+            const totalFilledAmount = takerAssetFilledAmount.add(orderTakerAssetFilledAmount);
+            if (totalFilledAmount.greaterThan(MAX_UINT256)) {
+                throw overflowErrorForSendTransaction;
+            }
+            // Note(albrow): In the Solidity implementation, if orderHash
+            // overflows bytes32, the return value is always 0.
```

Comment:
Solidity has no overflows for `bytes32`. Maybe you are testing the input encoders of the Javascript wrappers? But these don't need testing.